addons/web/static/src/search/search_bar/search_bar.js

```diff
@@ -236,6 +231,21 @@ export class SearchBar extends Component {
         return items;
     }
 
+    getPreposition(searchItem) {
+        const [, fieldType] = this.getFieldInfos(searchItem);
+        return _t(["date", "datetime"].includes(fieldType) ? "at" : "for");
```

Comment:
I know this line was just moved but if this works it works by accident, the terms cannot be extracted from this, they should be translated individually
```suggestion
        return ["date", "datetime"].includes(fieldType) ? _t("at") : _t("for");
```